### ISSUE TITLE
Clean up GitHub Actions jobs

### ISSUE THREAD

**Issue by freshgum-bubbles** [freshgum-bubbles/typedi#23]
Currently the GitHub Actions CI workflow is... [pretty ugly](https://github.com/freshgum-bubbles/typedi/blob/c1d294acd5aa1a226465269651098a574ca5cb03/.github/workflows/continuous-integration-workflow.yml).
We copy-paste the Node+PNPM installer pretty much everywhere.

I'd _really_ appreciate help from someone who actually knows how to use GitHub Actions.

https://github.com/freshgum-bubbles/typedi/blob/c1d294acd5aa1a226465269651098a574ca5cb03/.github/workflows/continuous-integration-workflow.yml#L17-L38

**Comment by rdonadono** [comment]
Hi,

Is your idea to optimize the installation phase of Node and PNPM and make it reusable in the other GH Actions?

I'll fork the project and take a look at it, if I solve it I'll open a PR for you.

**Comment by freshgum-bubbles** [comment]
> Is your idea to optimize the installation phase of Node and PNPM and make it reusable in the other GH Actions?

Yeah, pretty much :-)  I tried to do it myself (something something "Reusable workflows") but ended up baffled by the whole thing.  It'd be really nice if there was a way to test Actions locally :/

#22, #15 and quite a few commits to the Workflows was as far as I got before the current, copy-paste situation.  

If you're interested, there's another problem with Actions here which I don't have the wherewithall to fix: each one seems to run *twice* for pull requests (see https://github.com/freshgum-bubbles/typedi/pull/114 as an example).  I'm sure that's burned through quite a lot of my minutes already, but alas :(
